src/main/webapp/app/overview/course-statistics/course-statistics.component.ts

```diff
@@ -424,6 +482,38 @@ export class CourseStatisticsComponent implements OnInit, OnDestroy {
         this.totalRelativeScore = this.calculateTotalScoreForTheCourse(RELATIVE_SCORE);
     }
 
+    calculateReachableScores() {
+        const quizzesTotalCurrentMaxScore = this.calculateScoreTypeForExerciseType(ExerciseType.QUIZ, REACHABLE_SCORE);
+        const programmingExerciseTotalCurrentMaxScore = this.calculateScoreTypeForExerciseType(ExerciseType.PROGRAMMING, REACHABLE_SCORE);
+        const modelingExerciseTotalCurrentMaxScore = this.calculateScoreTypeForExerciseType(ExerciseType.MODELING, REACHABLE_SCORE);
+        const textExerciseTotalCurrentMaxScore = this.calculateScoreTypeForExerciseType(ExerciseType.TEXT, REACHABLE_SCORE);
+        const fileUploadExerciseTotalCurrentMaxScore = this.calculateScoreTypeForExerciseType(ExerciseType.FILE_UPLOAD, REACHABLE_SCORE);
+        const reachableScores = {};
+        reachableScores[ExerciseType.QUIZ] = quizzesTotalCurrentMaxScore;
+        reachableScores[ExerciseType.PROGRAMMING] = programmingExerciseTotalCurrentMaxScore;
+        reachableScores[ExerciseType.MODELING] = modelingExerciseTotalCurrentMaxScore;
+        reachableScores[ExerciseType.TEXT] = textExerciseTotalCurrentMaxScore;
+        reachableScores[ExerciseType.FILE_UPLOAD] = fileUploadExerciseTotalCurrentMaxScore;
+        this.reachableScores = reachableScores;
+        this.reachableScore = this.calculateTotalScoreForTheCourse(REACHABLE_SCORE);
+    }
+
+    calculateCurrentRelativeScores(): void {
+        const quizzesCurrentRelativeScore = this.calculateScoreTypeForExerciseType(ExerciseType.QUIZ, CURRENT_RELATIVE_SCORE);
```

Comment:
Why do you need all those constants? As far as I can see you you just assign them below?